File: services/typescript/elysia/src/routes/handlers/invalid_method.ts
import type { Context } from "elysia";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export function invalidMethodHandler({ request, path, set }: Context) {
  getLogger().warn({ method: request.method, path }, "Invalid method");
  return sendError(set, AppErrors.InvalidMethod);
}

File: services/python/sanic/src/routes/handlers/invalid_method.py
import structlog
from sanic import Request
from sanic.response import JSONResponse
from sanic.exceptions import MethodNotAllowed

from routes.errors import AppErrors, send_error


async def invalid_method_handler(
    request: Request, _exception: MethodNotAllowed
) -> JSONResponse:
    structlog.get_logger().warn(
        "Invalid method", method=request.method, path=request.path
    )

    return send_error(AppErrors.InvalidMethod)
